tests/python_package_test/test_engine.py

```diff
@@ -764,6 +764,41 @@ def test_early_stopping():
     assert 'binary_logloss' in gbm.best_score[valid_set_name]
 
 
+@pytest.mark.parametrize('use_valid', [True, False])
+def test_early_stopping_ignores_training_set(use_valid):
+    x = np.linspace(-1, 1, 100)
+    X = x.reshape(-1, 1)
+    y = x**2
+    X_train, X_valid = X[:80], X[80:]
+    y_train, y_valid = y[:80], y[80:]
+    train_ds = lgb.Dataset(X_train, y_train)
+    valid_ds = lgb.Dataset(X_valid, y_valid)
+    valid_sets = [train_ds]
+    valid_names = ['train']
+    if use_valid:
+        valid_sets.append(valid_ds)
+        valid_names.append('valid')
+    eval_result = {}
+
+    def train_fn():
+        return lgb.train(
+            {'num_leaves': 5},
```

Comment:
```suggestion
            params={'num_leaves': 5, 'num_iterations': 10},
```

To save some CI time, would you consider reducing the number of boosting rounds? ([the default is 100](https://lightgbm.readthedocs.io/en/latest/Parameters.html#num_iterations))

Reply:
oh sorry, missed that!